### PR TITLE
String interpolation updates

### LINKED ISSUE #27209: Tutorial: Write a custom string interpolation handler – couple of small mistakes
### # 1
```csharp
[InterpolatedStringHandler]
public ref struct LogInterpolatedStringHandler
{
    …
    // Add the receiver argument:
    …
```
I guess `// Add the receiver argument:` is forgotten since [Improved Interpolated Strings](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.defaultinterpolatedstringhandler?view=net-6.0) discusses declaratively _receiver_ term. 

### # 2
```csharp
public LogInterpolatedStringHandler(int literalLength, int formattedCount, LogLevel level, Logger logger, out bool isEnabled)
{
    enabled = logger.EnabledLevel >= level;
    Console.WriteLine($"\tliteral length: {literalLength}, formattedCount: {formattedCount}");
    if (enabled)
    {
        builder = new StringBuilder(literalLength);
    }
    isEnabled = enabled;
}
```

In fact there are 2 failures:
* `logger` and `level` parameters are switched so declaration `[InterpolatedStringHandlerArgument("", "level")] LogInterpolatedStringHandler builder` does not work with it.
* `LogInterpolatedStringHandler` is declared as `struct` so `builder` must be definitely assigned when .ctor returns. So either it must be changed to `class` or used something like `builder = enabled ? new () : null;`.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 99599c47-14ba-6e35-9b63-daa928f764bb
* Version Independent ID: e80a06db-bfbc-29f1-1cca-248460e523d3
* Content: [Explore string interpolation handlers](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/improved-interpolated-strings)
* Content Source: [docs/csharp/whats-new/tutorials/interpolated-string-handler.md](https://github.com/dotnet/docs/blob/main/docs/csharp/whats-new/tutorials/interpolated-string-handler.md)
* Product: **dotnet-csharp**
* Technology: **csharp-whats-new**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #27271: Nullability of `builder` when conditionally setting it
The part where the page says we can have an `out bool isEnabled` should have address the null dereference errors that are now produced, e.g. the constructor could instead be:

```csharp

```
    public LogInterpolatedStringHandler(int literalLength, int formattedCount, Logger logger, LogLevel logLevel, out bool isEnabled)
    {
        isEnabled = enabled = logger.EnabledLevel >= logLevel;
        Console.WriteLine($"\tliteral length: {literalLength}, formattedCount: {formattedCount}");
        builder = isEnabled ? new StringBuilder(literalLength) : null;
    }
---

This also now means that the `enabled` field is no longer used.  And if that field is no longer used, then we can revert the signatures of the `Append*` methods from `bool` back to `void` and remove the checks for `enabled`.

#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 99599c47-14ba-6e35-9b63-daa928f764bb
* Version Independent ID: e80a06db-bfbc-29f1-1cca-248460e523d3
* Content: [Explore string interpolation handlers](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/interpolated-string-handler)
* Content Source: [docs/csharp/whats-new/tutorials/interpolated-string-handler.md](https://github.com/dotnet/docs/blob/main/docs/csharp/whats-new/tutorials/interpolated-string-handler.md)
* Product: **dotnet-csharp**
* Technology: **csharp-whats-new**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Review and fix open issues on the string interpolation tutorial

Fixes #27209

- Remove stray comment.
- Ensure parameters are ordered consistently across all snippet versions.

Fixes #27271 

- A little refactoring and explanation.

### CODE REVIEW

**Comment by BillWagner** [comment]
ping @adegeo 

Once again, each fix is in a separate commit for easier review.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Comment by Uzivatel919** [comment]
Fine new paragraph, @BillWagner 

> That change means you can remove the `enabled` field. Then, you can change the return type of `AppendLiteral` and `AppendFormattable` to `void`.

but there is little typo – AppendFormatt**able** should be AppendFormatt**ed**.

**Comment by adegeo** [comment]
@Uzivatel919 these comments in completed PRs may easily go unnoticed. It's best to open a new issue or edit the file and open a PR.

File: docs/csharp/whats-new/tutorials/interpolated-string-handler.md [hunk 1]
```diff
@@ -169,6 +169,7 @@ You can make one final update to the handler's constructor that improves efficie
 
 :::code language="csharp" source="./snippets/interpolated-string-handler/logger-v4.cs" id="UseOutParameter":::
 
+That change means you can remove the `enabled` field. Then, you can change the return type of `AppendLiteral` and `AppendFormattable` to `void`.
```
Comment (Uzivatel919) [inline docs/csharp/whats-new/tutorials/interpolated-string-handler.md:172]: _AppendFormatt**able**_ should be _AppendFormatt**ed**_.

**Comment by Uzivatel919** [comment]
> @Uzivatel919 these comments in completed PRs may easily go unnoticed. It's best to open a new issue or edit the file and open a PR.

It’s fixed anyway.

**Pull request merged**